HW6.py

```diff
@@ -0,0 +1,56 @@
+import xml.etree.ElementTree as ET
+import re
+
+
+def parse_and_remove(filename, path):
+    path_parts = path.split('/')
+    doc = ET.iterparse(filename, ('start', 'end'))
+    # Skip root element
+    next(doc)
+    tag_stack = []
+    elem_stack = []
+    for event, elem in doc:
+        if event == 'start':
+            tag_stack.append(elem.tag)
+            elem_stack.append(elem)
+        elif event == 'end':
+            if tag_stack == path_parts:
+                yield elem
+            try:
+                tag_stack.pop()
+                elem_stack.pop()
+            except IndexError:
+                pass
+#code for task 1 Country name and goverment with out duplicates
+
+lines_seen = set()
+country_government = []
+countries = parse_and_remove('mondial-3.0.xml', 'country')
+for country in countries:
+    government = country.attrib['government'].split(' ')[0]
+    name = country.attrib['name']
+    new_country = re.compile(r"[A-Z]")
+    new_name = new_country.findall(name)
+
+    for n in new_name:
+        if country not in lines_seen:
+            lines_seen.add(country)
+            print(name, ':', government, end=',')
```

Comment:
1. The results are incorrect.
Actual and expected data:
![image](https://user-images.githubusercontent.com/34323554/73608747-37817800-45cf-11ea-988c-df870781c3db.png)

2. The task was to _return all distinct governments_. 
So the code should be like:
```
unique_governments = []
countries = parse_and_remove('mondial-3.0.xml', 'country')
for country in countries:
    government = country.attrib['government'].lstrip()
    if government not in unique_governments:
        unique_governments.append(government)
print(unique_governments)
```